plugins/sql_loader/tasks.py

```diff
@@ -178,29 +164,30 @@ def second_task_execution(
     )
     df = None
     if custom_query:
-        table_query = db_query
+        db_query = db_query.strip()
+        table_query = db_query if db_query.startswith("SELECT") else ""
     else:
-        table_query = f"SELECT {columns_list} FROM {table_name}"
-
-    print(f"table_query: {table_query}")
+        table_query = (
+            f"SELECT {columns_list} FROM {table_name}" if columns_list != "" else ""
+        )
+        retrieval_only = True
 
     if table_query != "":
-        if columns_list != "":
-            if limit is not None:
-                table_query = f"SELECT * FROM ({table_query}) AS temp LIMIT {limit}"
-            df = db_manager.get_query_as_dataframe(table_query)
-
-            # Check if given attribute can be used as a unique identifier
-            # If so, use attribute
-            if id_attribute in df and len(df[id_attribute].unique()) == len(
-                df[id_attribute]
-            ):
-                df["ID"] = df[id_attribute]
-            # If not, use indices
-            else:
-                df["ID"] = list(range(df.shape[0]))
-            df["href"] = [get_href(db_host, db_port, db_database)] * df.shape[0]
-    else:
+        if limit is not None and isinstance(limit, int) and limit > 0:
+            table_query = table_query.rstrip(";")
```

Comment:
this will not work if the `;` is followed by a comment (e.g. `;  -- end of line`) but at this point (and for this kind of software) I think we have done enough about the safety of this plugin.

Reply:
Then I will keep it as is and write a warning in the description.